3.5/programs-arangobackup-limitations.md

```diff
@@ -0,0 +1,155 @@
+
+Also note that in a cluster each RocksDB instance will be backed up
+individually and hence the overall storage space will be the sum of all
+RocksDB instances (i.e., data which is replicated between instances will
+not be de-duplicated for performance reasons).
+
+Global Transaction Lock
+-----------------------
+
+In order to be able to create consistent hot backups, it is mandatory to get
+a very brief global transaction lock across the entire installation.
+In single server deployments constant invocation of very long running
+transactions could prevent that from ever happening during a timeout period.
+The same holds true for clusters, where this lock must now be obtained on all
+database servers at the same time.
+
+Especially in the cluster the result of these successively longer tries to
+obtain the global transaction lock might become visible in periods of apparent
+dead time. Locks might be obtained on some machines and and not on others, so
+that the process has to be retried over and over. Every unsuccessful try would
+then lead to the release of all partial locks.
+
+ArangoSearch not Supported Yet
+------------------------------
+
+ArangoSearch views are not backed up and thus not restored yet.
+Therefore, views have to be dropped and recreated after a restore.
+This happens automatically in the background, but in particular in the
+presence of large amounts of data, the recreation of the ArangoSearch
+indexes can take some time after the restore. It is planned to rectify
+this limitation in one of the next releases.
+
+Note furthermore that a running query with views can prevent a restore
+operation from happening whilst the query is running.
+
+Windows not Supported Yet
+-------------------------
+
+The first release of the hot backup feature (3.5.1) does not support
+the Windows version of ArangoDB. 
+
+Services on Single Server
+-------------------------
+
+On a single server the installed services are not backed up and are
+therefore also not restored. This is because in single server mode
+the service installation is done locally in the file system and does not
+track the information in the `_apps` collection.
+
+In a cluster, the coordinators will after a restore eventually restore
+the state of the services from the `_apps` collection.
+
+Encryption at Rest
+------------------
+
+Currently, the hot backup simply takes a snapshot of the database files.
+If one is using encryption at rest, then the backed up files will be
+encrypted, with the encryption key that was used in the
+instance which created the backup.
+
+Such an encrypted backup can only be restored to an instance using the
+same encryption key.
+
+Replication and Hot Backup
+--------------------------
+
+Hot backups are not automatically replicated between instances. This is
+true both for the active/failover setup with 2 (or more) single servers,
+and for the data center to data center replication between clusters.
+Simply take hot backups on all instances.
```

Comment:
Is there a chance of running into trouble if the passive side is behind the active side when both are restored?